challenges/headstorm-bec-jehillert/.notes/README - Applicant's Notes.txt

```diff
@@ -0,0 +1,213 @@
+TO DO
+———————————————————————————————————————————————————————————————————————————————
+  Make sure error from bodyparser indicates the correct information.
+
+  X Make sure content of rest api tutorial is not something that is going to change what you thought the assignment was about.
+
+  Provide API documentation.  Specifically, instructions on the form the data should be in (e.g., array or object), json, etc.
+
+  Changes to make to variable names:
+    sorted_list_val_id ==> sorted_val_id
+    unsorted_list_val_id ==> unsorted_val_id
+
+QUESTIONS
+———————————————————————————————————————————————————————————————————————————————
+  Hi Gianna,
+
+  Would it be possible to pass the following questions on to the developers in charge of the Back End Challenge?  Brief answers would be fine (and greatly appreciated)
+
+  Thanks much,
+
+  John
+
+  -----------------------------------------------
+
+  QUESTION 1: Should we store the data as a table row containing a JSON, a table of parsed number values (each number a row), or is there another preferred structure. My [imperfect] understanding is that a table of values is strongly preferred, at least in the context of relational databases.
+
+  QUESTION 2:  May we store the data in unsorted form, and handle sorting at the retrieval stage?  My understanding is that adding new values to the bottom rows of a table as the values come in, and then handling sorting when the data is needed, is preferred.
+
+  QUESTION 3:  Is the 500 number constraint only applicable to execution of the Post request or is the PATCH request supposed to conform to the same limitation?  In the latter case, should the code delete another member of the list every time a new one is inserted?
+
+  QUESTION 4: Can you clarify what is meant by "a PATCH endpoint at `/data` which inserts a random number into the list in the proper order which will be returned by the above POST api?"  This question ties in to 2nd and 3rd. Is this part of the instructions suggesting that the data should be maintained/stored at the back-end as an ordered list (Q2)?  Is the PATCH supposed to conclude by utilizing some part of the POST route? Also, the POST API suggests the list should stay at 500 members, but the effect of the PATCH would create 501 (Q3).
```

Comment:
In regards to Q3/Q4 I think that it would be acceptable to have the POST be restricted to 500 at a time, and then allow the user to add one value at a time with the PATCH request. I think that then the GET should return ALL the numbers in the db sorted from lowest to highest, even if there are more than 500. 

I think that this is probably an error in how we wrote the question, upon rereading, the PATCH part of the instruction does seem to conflict with the rules for the other two endpoints. Nice catch!